### ISSUE TITLE
Kubermatic Operator: Add Proxy Settings

### ISSUE THREAD

**Issue by xrstf** [kubermatic/kubermatic#4783]
**User Story**
The Helm chart allows to configure HTTP proxies. This should be possible via the Operator as well.

This is part of #4723.

**Acceptance criteria**
* [ ] All proxy-related settings from the chart are possible via KubermaticConfiguration.

**Comment by xrstf** [comment]
I seem to have mixed up what "proxy settings" actually mean. @kdomanski can you clarify this requirement?

**Comment by erdii** [comment]
in my understanding, proxy settings for kubermatic are:
* (propably differing, marked with capital letters) tuples of http_proxy, https_proxy, no_proxy envvars
* for:
  * (A) everything the kubermatic-operator needs to talk outside of the cluster
  * (A) everything the kubermatic-operator sets up and that needs to talk outside of the cluster
  * (A - ability to toggle on/off) talking to a seed-cluster that is possibly only reachable trough the proxy
  * (B) customer-cluster control-planes, which need to be supplied with proxy settings (this includes essential cluster-components, like machine-controller, cloud-controller-manager, etc)
  * (C) customer-cluster machine-controller needs to supply proxy settings to the created machines so they can reach the internet

* it should be possible that A, B, C are different proxy settings because these components can sit behind different proxies
*  it would be very nice to automate the no_proxy envvar like kubeone does (automatically excluding in-cluster-cidrs,cluster domain, services that need to be reached within the cluster)

---

Basically every component that potentially communicates with external components needs the possibility to sit behind a proxy:
- KubeOne components (and seed cluster itself) ✔
- Kubermatic operator ❓
- Kubermatic controllers which are talking with the cloud providers ❓
- Kubernetes/OpenShift control plane components ❓
- Worker nodes ❓(machine-controller supports that - tested with kubeone)

**Comment by kdomanski** [comment]
> I seem to have mixed up what "proxy settings" actually mean. @kdomanski can you clarify this requirement?

@xrstf The same old thing with no internet access except through a configured http proxy.

**Comment by kdomanski** [comment]
It seems that the bulk of it is configured at seed level. Apart from that the operator needs to be able to use image from a local docker image mirror. See also https://github.com/kubermatic/kubermatic/blob/master/api/hack/ci/ci-deploy-offline.sh

**Comment by eqrx** [comment]
Additionally we do not know which components of kubermatic need to communicate with some external services. For example dex needs to reach the authenticator it connects to, so dex needs to be configured to use the proxy outbound if the authenticator is not directly reachable. 

But for all the other kubermatic components.... no idea what would need to communicate with what.

We currently use PodPresets to override settings in kubermatic but that is very "meh".
